src/libraries/Common/tests/System/Security/Cryptography/X509Certificates/CertificateAuthority.cs

```diff
@@ -823,14 +849,38 @@ internal static void BuildPrivatePki(
             // default to client
             extensions ??= new X509ExtensionCollection() { s_eeConstraints, s_eeKeyUsage, s_tlsClientEku };
 
-            using (RSA rootKey = RSA.Create(keySize))
-            using (RSA eeKey = RSA.Create(keySize))
+            if (keyFactory is null)
+            {
+                // This could use any of the NC-hashes, but that complicates the code sharing for this file,
```

Comment:
```suggestion
                // This could use any of the non-cryptographic hashes, but that complicates the code sharing for this file,
```

As far as I know, North Carolina has not standardized a hash (this is my way of saying "It took me too long to figure out what NC meant)